### DISCUSSION TITLE
Allow attach meta function to enum

### DISCUSSION THREAD

**Discussion by codehz** [edubart/nelua-lang#97] [Ideas]
Like __tostring so we can print enum easily

**Comment by edubart** [comment]
Support for this was added in commit https://github.com/edubart/nelua-lang/commit/2c86ae0bd818c42f83d06067d779565797bf959c
Here is an example on how to use:

```lua
require 'string'

local MyEnum: type = @enum{A = 0, B}

function MyEnum.__tostring(value: MyEnum): string
  ## for i,field in ipairs(value.type.fields) do
  if value == #[field.value]# then
    return string.copy(#[value.type.nickname..'.'..field.name]#)
  end
  ## end
  return string.copy('Unknown')
end

print(MyEnum.A) -- outputs: MyEnum.A
```

Note that I've used `string.copy` because `tostring` must return a new allocated string, as mentioned in https://github.com/edubart/nelua-lang/issues/101
